File: backend/src/main/java/com/elms/service/ReportService.java
package com.elms.service;

import java.util.List;
import java.util.Map;

public interface ReportService {
    
    Map<String, Object> getLeaveSummary();
    
    Map<String, Object> getDepartmentSummary();
    
    Map<String, Object> getMonthlyReport(int year, int month);
    
    byte[] exportToCsv();
}

File: frontend/src/services/reportService.js
import api from './api';

export const reportService = {
  getLeaveSummary: async () => {
    const response = await api.get('/reports/leave-summary');
    return response.data;
  },

  getDepartmentSummary: async () => {
    const response = await api.get('/reports/department-summary');
    return response.data;
  },

  getMonthlyReport: async (year, month) => {
    const response = await api.get('/reports/monthly-report', {
      params: { year, month },
    });
    return response.data;
  },

  exportToCsv: async () => {
    const response = await api.get('/reports/export-csv', {
      responseType: 'blob',
    });
    return response.data;
  },
};
